sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricNameFilter.java

```diff
@@ -54,7 +54,6 @@ public static MetricNameFilter named(String namespace, String name) {
   public static MetricNameFilter named(Class<?> namespace, String name) {
     checkNotNull(namespace, "Must specify a inNamespace");
     checkNotNull(name, "Must specify a name");
-    return new AutoValue_MetricNameFilter(namespace.getSimpleName(), name);
+    return new AutoValue_MetricNameFilter(namespace.getName(), name);
```

Comment:
This seems to be a positive change, but I'm not sure it is needed for this PR.

I'm assuming this change was done to match https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricName.java#L45-L45

There seems to be no test coverage for this. I'd add tests as well (Although again, probably in a separate PR).